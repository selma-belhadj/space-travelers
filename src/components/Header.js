import React from 'react';
import logo from '../images/logo.png';

function Header() {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Rockets',
    },
    {
      id: 2,
      path: '/Dragons',
      text: 'Dragons',
    },
    {
      id: 3,
      path: '/Missions',
      text: 'Missions',
    },
    {
      id: 4,
      path: '/Profile',
      text: 'My Profile',
    },
  ];
  return (
    <header>
      <div>
        <img src={logo} alt="logo" style={{ width: '4.5rem', height: '4.5rem' }} />
        <h1>Space Travelers&apos; Hub</h1>
      </div>
      <nav>
        {links.map((link) => (
          <NavLink
            key={link.id}
            to={link.path}
          >
            {link.text}
          </NavLink>
        ))}
      </nav>
    </header>
  )
}

export default Header
