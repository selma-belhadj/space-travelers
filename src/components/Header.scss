$medium: 900px;

header {
  background-color: white;
  position: fixed;
  inset: 0;
  height: 90px;
  display: flex;
  justify-content: space-between;
  align-items: center;
  box-shadow: 2px 2px 8px 0 darkgrey;
  z-index: 10;

  .logo {
    display: flex;
    align-items: center;
    gap: 1.5rem;
  }

  nav {
    display: flex;
    gap: 10px;

    a {
      text-decoration: none;
      color: darkgrey;
    }

    a:hover {
      color: black;
    }

    a:last-child {
      border-left: 1px solid darkgrey;
      padding-left: 10px;
    }

    .active {
      color: black;
    }
  }
}

@media only screen and (max-width: $medium) {
  .logo {
    h1 {
      font-size: 1.5rem;
    }
  }

  header {
    flex-direction: column;
    gap: 10px;
    height: 120px;
    justify-content: center;
  }
}
