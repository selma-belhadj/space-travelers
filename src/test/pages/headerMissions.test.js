import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import Header from '../../components/Header';
import Missions from '../../pages/MissionsPage';

describe('Test header component', () => {
  it('header renders correctly', () => {
    const tree = render(
      <Router>
        <Header />
      </Router>,
    );
    expect(tree).toMatchSnapshot();
  });
});

it('Test Missions renders correctly', () => {
  const tree = render(
    <Provider store={store}>
      <Missions />
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
