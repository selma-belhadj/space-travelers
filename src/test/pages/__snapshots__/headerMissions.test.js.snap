// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Missions renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="spacing"
        id="missions-page"
      >
        <table>
          <thead>
            <tr>
              <th
                class="mission-name"
              >
                Mission
              </th>
              <th
                class="mission-description"
              >
                Description
              </th>
              <th
                class="mission-status"
              >
                Status
              </th>
              <th
                aria-label="mission action"
                class="mission-action"
              />
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="spacing"
      id="missions-page"
    >
      <table>
        <thead>
          <tr>
            <th
              class="mission-name"
            >
              Mission
            </th>
            <th
              class="mission-description"
            >
              Description
            </th>
            <th
              class="mission-status"
            >
              Status
            </th>
            <th
              aria-label="mission action"
              class="mission-action"
            />
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test header component header renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="spacing"
      >
        <div
          class="logo"
        >
          <img
            alt="logo"
            src="logo.png"
          />
          <h1>
            Space Travelers' Hub
          </h1>
        </div>
        <nav>
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            Rockets
          </a>
          <a
            class=""
            href="/Dragons"
          >
            Dragons
          </a>
          <a
            class=""
            href="/Missions"
          >
            Missions
          </a>
          <a
            class=""
            href="/Profile"
          >
            My Profile
          </a>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="spacing"
    >
      <div
        class="logo"
      >
        <img
          alt="logo"
          src="logo.png"
        />
        <h1>
          Space Travelers' Hub
        </h1>
      </div>
      <nav>
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          Rockets
        </a>
        <a
          class=""
          href="/Dragons"
        >
          Dragons
        </a>
        <a
          class=""
          href="/Missions"
        >
          Missions
        </a>
        <a
          class=""
          href="/Profile"
        >
          My Profile
        </a>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
